api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -238,6 +238,52 @@ message HttpConnectionManager {
     google.protobuf.BoolValue enabled = 3;
   }
 
+  // Transformations that apply to path headers. The policy to apply these
+  // transformations can be tuned by the configuration field. Transformations
+  // can be applied internally only, so that the normalized header is used in
+  // matching, routing, and HTTP filters while the original path header is
+  // forwarded upstream, or additionally apply to the forwarded *:path* header.
+  message PathNormalizationOptions {
+    // Defines a policy to apply normalization.  Path rewrites :ref:`regex_rewrite
+    // <envoy_api_field_route.RouteAction.regex_rewrite>` or :ref:`prefix_rewrite
+    // <envoy_api_field_route.RouteAction.prefix_rewrite>` will apply to the
+    // *:path* header destined for the upstream.
+    enum NormalizationPolicy {
+      // Normalizations apply internally before any processing of requests by HTTP
+      // filters, routing, and matching, *and* will affect the upstream *:path* header
+      // as well.
+      FORWARD = 0;
+
+      // Normalizations only apply internally. The original *:path* header will
+      // be sent upstream *unless* the path has been mutated by a filter. Any
+      // regex rewrites that the router performs will be on the upstream *:path*
+      // header.
+      INTERNAL = 1;
+    }
+
+    NormalizationPolicy policy = 1;
+
+    // Should paths be normalized according to RFC 3986 before any processing of
+    // requests by HTTP filters or routing? For paths that fail this check, Envoy
+    // will respond with 400 to paths that are malformed. This defaults to
+    // true. When not specified, this value may be overridden by the runtime
+    // variable
+    // :ref:`http_connection_manager.normalize_path<config_http_conn_man_runtime_normalize_path>`.
+    // See `Normalization and Comparison
+    // <https://tools.ietf.org/html/rfc3986#section-6>`_ for details of
+    // normalization.  Note that Envoy does not perform `case normalization
+    // <https://tools.ietf.org/html/rfc3986#section-6.2.2.1>`_
+    google.protobuf.BoolValue normalize_path = 2;
+
+    // Determines if adjacent slashes in the path are merged into one before any
+    // processing of requests by HTTP filters or routing. Without setting this
+    // option, incoming requests with path `//dir///file` will not match against
+    // route with `prefix` match set to `/dir`. Defaults to `false`. Note that
```

Comment:
According to the [API_STYLE guide](https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/api/STYLE.md#api-style-guidelines), the wrapped scalar type should be used in cases that the default value isn't compatible with proto3 default (false in this case). Is there an explicit reason for using the wrapper rather than `bool`?

Reply:
This allows us to change defaults in future..